scripts/Engineering/EnginX.py

```diff
@@ -0,0 +1,381 @@
+        elif cropped == "spectra":
+            create_calibration_cropped_file(True, crop_on, crop_name,
+                                            van_curves_file, van_int_file, ceria_run, calibration_directory,
+                                            van_run, calibration_general)
+    else:
+        create_calibration_files(van_curves_file, van_int_file, ceria_run, calibration_directory, van_run,
+                                 calibration_general)
+
+
+# create and save a cropped calibration file
+def create_calibration_cropped_file(use_spectrum_number, spec_nos, crop_name, curve_van, int_van, ceria_run, cal_dir,
+                                    van_run, cal_gen):
+    van_curves_ws, van_integrated_ws = load_van_files(curve_van, int_van)
+    ceria_ws = simple.Load(Filename="ENGINX" + ceria_run, OutputWorkspace="eng_calib")
+    param_tbl_name = None
+    bank = None
+    # check which cropping method to use
+    if use_spectrum_number:
+        if crop_name is None:
+            param_tbl_name = "cropped"
+        else:
+            param_tbl_name = crop_name
+        # run the calibration cropping on spectrum number
+        output = simple.EnggCalibrate(InputWorkspace=ceria_ws, VanIntegrationWorkspace=van_integrated_ws,
+                                      VanCurvesWorkspace=van_curves_ws, SpectrumNumbers=str(spec_nos),
+                                      FittedPeaks=param_tbl_name,
+                                      OutputParametersTableName=param_tbl_name)
+        # get the values needed for saving out the .prm files
+        difc = [output.DIFC]
+        tzero = [output.TZERO]
+        difa = [output.DIFA]
+        save_calibration(ceria_run, van_run, ".prm", cal_dir, [param_tbl_name], difc, tzero, "all_banks", cal_gen)
+        save_calibration(ceria_run, van_run, ".prm", cal_dir, [param_tbl_name], difc, tzero,
+                         "bank_{}".format(param_tbl_name), cal_gen)
+    else:
+        # work out which bank number to crop on, then calibrate
+        if spec_nos.lower() == "north":
+            param_tbl_name = "engg_calibration_bank_1"
+            bank = 1
+        elif spec_nos.lower() == "south":
+            param_tbl_name = "engg_calibration_bank_2"
+            bank = 2
+        output = simple.EnggCalibrate(InputWorkspace=ceria_ws, VanIntegrationWorkspace=van_integrated_ws,
+                                      VanCurvesWorkspace=van_curves_ws, Bank=str(bank), FittedPeaks=param_tbl_name,
+                                      OutputParametersTableName=param_tbl_name)
+        # get the values needed for saving out the .prm files
+        difc = [output.DIFC]
+        tzero = [output.TZERO]
+        difa = [output.DIFA]
+        save_calibration(ceria_run, van_run, ".prm", cal_dir, [spec_nos], difc, tzero, "all_banks", cal_gen)
+        save_calibration(ceria_run, van_run, ".prm", cal_dir, [spec_nos], difc, tzero, "bank_{}".format(spec_nos),
```

Comment:
All the places `save_calibration` is called use ".prm" as the extension, so why have it as an argument?